spanner_orm/field.py

```diff
@@ -43,23 +43,43 @@ def grpc_type() -> spanner_v1.Type:
   def validate_type(value: Any) -> None:
     raise NotImplementedError
 
+  @staticmethod
+  def matches(type: str) -> bool:
+    raise NotImplementedError
+
+  @staticmethod
+  def support_length() -> bool:
+    raise NotImplementedError
+
 
 class Field(object):
   """Represents a column in a table as a field in a model."""
 
   def __init__(self,
                field_type: Type[FieldType],
                nullable: bool = False,
-               primary_key: bool = False):
+               primary_key: bool = False,
+               length: int = 0):
     self.name = None
     self._type = field_type
     self._nullable = nullable
     self._primary_key = primary_key
+    self._length = length
+
+    if not self._type.support_length() and self._length:
+      raise error.ValidationError('length can not be set on field {}'.format(
+          self._type))
+    
+    if self._length < 0:
+      raise error.ValidationError('length can not be less than zero')
 
   def ddl(self) -> str:
+    base_ddl = self._type.ddl()
+    if self._length:
+      base_ddl = base_ddl.replace('(MAX)', f'({self._length})')
```

Comment:
(nit) String replacement seems a bit brittle to me. Would you mind changing it so we can do `self._type.ddl(length=self._length)` here instead of needing to use string replacement?